salttesting/jenkins.py

```diff
@@ -245,21 +247,45 @@ def build_pillar_data(options, convert_to_yaml=True):
     if options.test_git_commit is not None:
         pillar['test_git_commit'] = pillar['repo_clone_rev'] = options.test_git_commit
     if options.test_git_url is not None:
-        pillar['test_git_url'] = pillar['repo_clone_url'] = options.test_git_url
+        # Due to quoting issues in Linux shelling out to Windows using WinEXE
+        # it is necessary to wrap some parameters in some crazy quotes '"'"'
+        # But, it's being converted to YAML at the end, so I don't know what's
+        # going to happen.
+        if getattr(options, 'windows', False):
+            q_url = '"\'"\'{0}\'"\'"'.format(options.test_git_url)
+            pillar['test_git_url'] = pillar['repo_clone_url'] = q_url
+        else:
+            pillar['test_git_url'] = pillar['repo_clone_url'] = options.test_git_url
     if options.bootstrap_salt_url is not None:
-        pillar['bootstrap_salt_url'] = options.bootstrap_salt_url
+        if getattr(options, 'windows', False):
+            q_url = '"\'"\'{0}\'"\'"'.format(options.bootstrap_salt_url)
+            pillar['bootstrap_salt_url'] = q_url
+        else:
+            pillar['bootstrap_salt_url'] = options.bootstrap_salt_url
     if options.bootstrap_salt_commit is not None:
         pillar['bootstrap_salt_commit'] = options.bootstrap_salt_commit
     if options.salttesting_namespec is not None:
         pillar['salttesting_namespec'] = options.salttesting_namespec
 
     # Build package pillar data
     if options.package_source_dir:
-        pillar['package_source_dir'] = options.package_source_dir
+        if getattr(options, 'windows', False):
+            q_url = '"\'"\'{0}\'"\'"'.format(options.package_source_dir)
+            pillar['package_source_dir'] = q_url
+        else:
+            pillar['package_source_dir'] = options.package_source_dir
     if options.package_build_dir:
-        pillar['package_build_dir'] = options.package_build_dir
+        if getattr(options, 'windows', False):
+            q_url = '"\'"\'{0}\'"\'"'.format(options.package_build_dir)
+            pillar['package_build_dir'] = q_url
+        else:
+            pillar['package_build_dir'] = options.package_build_dir
     if options.package_artifact_dir:
-        pillar['package_artifact_dir'] = options.package_artifact_dir
+        if getattr(options, 'windows', False):
+            q_url = '"\'"\'{0}\'"\'"'.format(options.package_artifact_dir)
+            pillar['package_artifact_dir'] = q_url
+        else:
+            pillar['package_artifact_dir'] = options.package_artifact_dir
```

Comment:
Changed to run a for loop on the pillar dict to modify entries that contain `_dir` and `_url`